studio/components/interfaces/TableGridEditor/TableDefinition.tsx

```diff
@@ -1,28 +1,72 @@
-import { useRef } from 'react'
 import Editor from '@monaco-editor/react'
-import { timeout } from 'lib/helpers'
-import { useStore } from 'hooks'
 import { useProjectContext } from 'components/layouts/ProjectLayout/ProjectContext'
-import { useViewDefinitionQuery } from 'data/database/view-definition-query'
 import ShimmeringLoader from 'components/ui/ShimmeringLoader'
+import { useTableDefinitionQuery } from 'data/database/table-definition-query'
+import { useViewDefinitionQuery } from 'data/database/view-definition-query'
+import { ENTITY_TYPE } from 'data/entity-types/entity-type-constants'
+import { useStore } from 'hooks'
+import useEntityType from 'hooks/misc/useEntityType'
+import { timeout } from 'lib/helpers'
+import { observer } from 'mobx-react-lite'
+import { useMemo, useRef } from 'react'
+import { format } from 'sql-formatter'
 
 export interface ViewDefinitionProps {
-  name: string
+  id?: number
 }
 
-const ViewDefinition = ({ name }: ViewDefinitionProps) => {
+const ViewDefinition = ({ id }: ViewDefinitionProps) => {
```

Comment:
probably worth renaming to match the file name -> TableDefinition

Reply:
I meant to do this but forgot! Nice catch!